obspy/clients/eida/fetch.py

```diff
@@ -0,0 +1,950 @@
+``"A"`` for a seismic event around 2010-02-27 07:00 (UTC). Optionally add
+``"-v"`` for verbosity. Resulting Mini-SEED data will be written to file
+``"data.mseed"``.
+
+.. code-block:: bash
+
+    $ %(prog)s -N '*' -S 'A*' -L '*' -C 'LHZ' \
+-s "2010-02-27T07:00:00Z" -e "2010-02-27T08:00:00Z" -v -o data.mseed
+
+The above request is anonymous and therefore restricted data will not be
+included. To include restricted data, use a file containing a token obtained
+from an EIDA authentication service and/or a CSV file with username and
+password for each node not implementing the EIDA auth extension.
+
+.. code-block:: bash
+
+    $ %(prog)s -a token.asc -c credentials.csv -N '*' -S 'A*' -L '*' -C 'LHZ' \
+-s "2010-02-27T07:00:00Z" -e "2010-02-27T08:00:00Z" -v -o data.mseed
+
+StationXML metadata for the above request can be requested using the following
+command:
+
+.. code-block:: bash
+
+    $ %(prog)s -N '*' -S 'A*' -L '*' -C 'LHZ' \
+-s "2010-02-27T07:00:00Z" -e "2010-02-27T08:00:00Z" -y station \
+-q level=response -v -o station.xml
+
+Multiple query parameters can be used:
+
+.. code-block:: bash
+
+    $ %(prog)s -N '*' -S '*' -L '*' -C '*' \
+-s "2010-02-27T07:00:00Z" -e "2010-02-27T08:00:00Z" -y station \
+-q format=text -q level=channel -q latitude=20 -q longitude=-150 \
+-q maxradius=15 -v -o station.txt
+
+Bulk requests can be made in ArcLink (-f), breq_fast (-b) or native FDSNWS POST
+(-p) format. Query parameters should not be included in the request file, but
+specified on the command line.
+
+.. code-block:: bash
+
+    $ %(prog)s -p request.txt -y station -q level=channel -v -o station.xml
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import sys
+import time
+import datetime
+import optparse
+import threading
+import socket
+import csv
+import re
+
+try:
+    # Python 3.2 and earlier
+    from xml.etree import cElementTree as ET  # NOQA
+
+except ImportError:
+    from xml.etree import ElementTree as ET  # NOQA
+
+try:
+    # Python 2.x
+    import Queue
+    import urllib2
+    import urlparse
+    import urllib
+
+except ImportError:
+    # Python 3.x
+    import queue as Queue
+    import urllib.request as urllib2
+    import urllib.parse as urlparse
+    import urllib.parse as urllib
+
+VERSION = "2016.140"
+
+GET_PARAMS = set(('net', 'network',
+                  'sta', 'station',
+                  'loc', 'location',
+                  'cha', 'channel',
+                  'start', 'starttime',
+                  'end', 'endtime',
+                  'service',
+                  'alternative'))
+
+POST_PARAMS = set(('service',
+                   'alternative'))
+
+
+class Error(Exception):
+    pass
+
+
+class AuthNotSupported(Exception):
+    pass
+
+
+class TargetURL(object):
+    def __init__(self, url, qp):
+        self.__scheme = url.scheme
+        self.__netloc = url.netloc
+        self.__path = url.path.rstrip('query').rstrip('/')
```

Comment:
I've been bitten by this sooo many times, but `(r)strip()` does not behave like one intuitively would expect and it might result in very hard to debug errors down the road. Please replace with:

```python
self.__path = re.sub("/?query$", "", url.path)
```